nomad/structs/acl.go

```diff
@@ -710,6 +710,16 @@ func (a *ACLAuthMethod) Copy() *ACLAuthMethod {
 	return c
 }
 
+// Canonicalize performs basic canonicalization on the ACL auth method object.
+func (a *ACLAuthMethod) Canonicalize() {
+	t := time.Now().UTC()
+
+	if a.CreateTime.IsZero() {
+		a.CreateTime = t
+		a.ModifyTime = t
```

Comment:
I think we will always want to update the modify time. We call this whenever an auth method is being upserted, so it will always be a modification to the state.